### PR TITLE
`assets_from_module` methods [2/2]

### PR DESCRIPTION
**Pull request by clairelin135**
Adds `group_name` and `asset_key_prefix` fields on `assets_from_module` methods. Refactors existing `AssetGroup.prefixed` to be in a helper function. For now, group names will be added to all `AssetsDefinition` instances (so will not apply to source assets).

Followups:
- Since this allows us to add group names fields on multi-assets, we should add `group_name` as a field on `multi_asset`
- `group_name` can also exist on `SourceAsset`?

### CODE REVIEW

File: python_modules/dagster/dagster/core/asset_defs/assets_from_modules.py [hunk 1]
```diff
@@ -71,7 +71,10 @@ def assets_and_source_assets_from_modules(
 
 
 def assets_from_modules(
-    modules: Iterable[ModuleType], extra_source_assets: Optional[Sequence[SourceAsset]] = None
+    modules: Iterable[ModuleType],
+    group_name: Optional[str] = None,
+    asset_key_prefix: Optional[str] = None,
+    extra_source_assets: Optional[Sequence[SourceAsset]] = None,
```
Comment (sryza) [inline python_modules/dagster/dagster/core/asset_defs/assets_from_modules.py:77]: I should have caught this the first time around: now that this just returns a list, I don't think there's any need for this `extra_source_assets` argument. The user could just concatenate this list to the list afterwards.

File: python_modules/dagster/dagster/core/asset_defs/assets_from_modules.py [hunk 2]
```diff
@@ -71,7 +71,10 @@ def assets_and_source_assets_from_modules(
 
 
 def assets_from_modules(
-    modules: Iterable[ModuleType], extra_source_assets: Optional[Sequence[SourceAsset]] = None
+    modules: Iterable[ModuleType],
+    group_name: Optional[str] = None,
+    asset_key_prefix: Optional[str] = None,
```
Comment (sryza) [inline python_modules/dagster/dagster/core/asset_defs/assets_from_modules.py:76]: Thoughts on just calling this `key_prefix` for brevity? Also, these need documentation.

**Review by sryza — CHANGES_REQUESTED** [review]
Looking good

File: python_modules/dagster/dagster/core/asset_defs/assets_from_modules.py (hunk 1 above)
Comment (clairelin135) [reply python_modules/dagster/dagster/core/asset_defs/assets_from_modules.py]: Got it--removed these.

File: python_modules/dagster/dagster/core/asset_defs/assets_from_modules.py (hunk 2 above)
Comment (clairelin135) [reply python_modules/dagster/dagster/core/asset_defs/assets_from_modules.py]: Changed this to `key_prefix` and added docstrings!

**Review by sryza — COMMENTED** [review]
Had a few more comments.

Also, while reviewing this PR, it occurred to me that `assets_from_package_module` should probably be `load_assets_from_package_module` , to be consistent with `load_assets_from_dbt_project` and to follow the general guideline of functions as verbs instead of nouns. My bad for recommending the wrong thing in the first place.

**Review by sryza — APPROVED** [review]
Nice!

**Pull request merged**